src/components/Card.js

```diff
@@ -12,34 +16,58 @@ export default class Card {
       .content
       .querySelector('.card')
       .cloneNode(true);
-
     return cardElement;
   }
 
+  _renderLikes() {
+    if (this._likeCountElement) {
+      this._likeCountElement.textContent = this._likes.length;
+    }
+  }
+
+  _setEventListeners() {
+    this._cardImage.addEventListener('click', () => {
+      this._handleCardClick(this._name, this._link);
+    });
+
+    this._likeButton.addEventListener('click', () => {
+      const isLiked = this._likeButton.classList.contains('card__like-button_active');
+      this._likeButton.classList.toggle('card__like-button_active'); 
+      localStorage.setItem(this._id, JSON.stringify(this._likeButton.classList.contains('card__like-button_active')));
+      this._handleLikeClick(this._id, !isLiked, this);
+    });
+    
+
+    this._deleteButton.addEventListener('click', () => {
+      this._handleDeleteClick(this._id, this._element);
+    });
+  }
+
+
+  updateLikes(newLikes) {
+    this._likes = newLikes;
+    this._renderLikes();
+  }
+
   getView() {
     this._element = this._getTemplate();
     this._cardImage = this._element.querySelector('.card__image');
     this._cardTitle = this._element.querySelector('.card__title');
     this._likeButton = this._element.querySelector('.card__like-button');
     this._deleteButton = this._element.querySelector('.card__trash-button');
+    this._likeCountElement = this._element.querySelector('.card__like-count');
 
     this._cardTitle.textContent = this._name;
     this._cardImage.src = this._link;
     this._cardImage.alt = this._name;
 
-    // Event Listeners
-    this._cardImage.addEventListener('click', () => {
-      this._handleCardClick(this._name, this._link);
-    });
-
-    this._likeButton.addEventListener('click', () => {
-      this._likeButton.classList.toggle('card__like-button_active');
-    });
+    const storedData = JSON.parse(localStorage.getItem(this._id));
```

Comment:
* `localStorage` can't be used for **likes handling**. You should check the card data. You have `isLiked` inside it. This is what is needed to deside whether to show `black` or `white` heart